pkg/kubelet/kubeletconfig/checkpoint/download_test.go

```diff
@@ -133,18 +143,39 @@ func TestRemoteConfigMapDownload(t *testing.T) {
 		expect Payload
 		err    string
 	}{
-		// object doesn't exist
-		{"object doesn't exist",
-			makeSource(&apiv1.NodeConfigSource{ConfigMapRef: &apiv1.ObjectReference{Name: "bogus", Namespace: "namespace", UID: "bogus"}}),
-			nil, "not found"},
-		// UID of downloaded object doesn't match UID of referent found via namespace/name
-		{"UID is incorrect for namespace/name",
-			makeSource(&apiv1.NodeConfigSource{ConfigMapRef: &apiv1.ObjectReference{Name: "name", Namespace: "namespace", UID: "bogus"}}),
-			nil, "does not match"},
-		// successful download
-		{"object exists and reference is correct",
-			makeSource(&apiv1.NodeConfigSource{ConfigMapRef: &apiv1.ObjectReference{Name: "name", Namespace: "namespace", UID: "uid"}}),
-			payload, ""},
+		{
+			desc: "object doesn't exist",
+			source: makeSource(&apiv1.NodeConfigSource{ConfigMap: &apiv1.ConfigMapNodeConfigSource{
+				Name:             "bogus",
+				Namespace:        "namespace",
+				UID:              "bogus",
+				KubeletConfigKey: "kubelet",
+			}}),
+			expect: nil,
+			err:    "not found",
+		},
+		{
+			desc: "UID is incorrect for namespace/name",
+			source: makeSource(&apiv1.NodeConfigSource{ConfigMap: &apiv1.ConfigMapNodeConfigSource{
+				Name:             "name",
+				Namespace:        "namespace",
+				UID:              "bogus",
+				KubeletConfigKey: "kubelet",
```

Comment:
Is there a reason why we dont use `kubeadmconstants.KubeletBaseConfigurationConfigMapKey` here?  Using "key" or something other than the actual key would make it clearer that we aren't relying on the actual key.  Or we should use the constant for the key if we are relying on that.

Reply:
1) Kubelet should never depend on kubeadm
2) I'm not sure I understand your question? Whatever `kubeletConfigKey` is set to determines where the Kubelet looks for the Kubelet config. There is no "actual key" - the choice of where to put the config in the ConfigMap is up to the user.

Reply:
I see, thanks for clarifying.